mycroft/skills/settings.py

```diff
@@ -93,18 +92,28 @@ def __init__(self, directory, name):
         # set file paths
         self._settings_path = join(directory, 'settings.json')
         self._meta_path = join(directory, 'settingsmeta.json')
-        self.is_alive = True
         self.loaded_hash = hash(str(self))
         self._complete_intialization = False
         self._device_identity = None
         self._api_path = None
         self._user_identity = None
         self.changed_callback = None
+        self._poll_timer = None
 
         # if settingsmeta exist
         if isfile(self._meta_path):
             self._poll_skill_settings()
 
+    def run_poll(self, _=None):
+        """Immediately poll the web for new skill settings"""
+        if self._poll_timer:
+            self._poll_timer.cancel()
+            self._poll_skill_settings()
+
+    def stop_polling(self):
+        if self._poll_timer:
+            self._poll_timer.cancel()
```

Comment:
The `cancel` will only work if the method is waiting to start executing _poll_skill_settings. Since the `_poll_skill_settings` does a couple of remote calls it might take some time to execute and there is a risk that the setting poll isn't stopped.